ceph-nvmeof.conf

```diff
@@ -15,6 +15,11 @@ addr = 127.0.0.1
 port = 5500
 enable_auth = False
 
+[discovery]
+
+addr =
+port = 4420
```

Comment:
I believe the standard port for discovery is 8009. Probably best to default to that. 

https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?&page=109#NVM_Express_Inc

Reply:
Yes, I know it.  I considered using `4420` after referring to many materials and multiple projects. Perhaps we can discuss which one is better.
1.  `4420` has been assigned for use by `NVMe over Fabrics`.  Personally, I think it is a more comman interface than `8009`. https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=4420
2. Most projects use `4420` as the default port instead of `8009`, such as `nvme-cli`, `spdk`, `wireshark`.
3. For nvme-cli, refer: https://github.com/linux-nvme/nvme-cli/blob/c2e36601f91cba2b7fe701d2a2a3a1cc8748f462/Documentation/nvme-discover.html#L1199
https://github.com/linux-nvme/nvme-cli/blob/c2e36601f91cba2b7fe701d2a2a3a1cc8748f462/Documentation/nvme-discover.html#L1218
4. All examples of discovery in spdk use `4420`.
5. Wireshark can only parse commands with port number `4420` as `nvme-tcp`, The command with port number `8009` cannot be parsed and can only be recognized as `TCP`. My code also referenced some designs from SPDK and Wireshark.
6. Of course, there are also some projects that believe that both 4420 and 8009 are acceptable, like `nvme-stas`.

Reply:
The projects I have listed may not simply be Discovery controllers, but NVMe over fabrics that include Discovery controllers, IO controllers and admin controllers, so 4420 is used. My project is purely a discovery controller, perhaps using 8009 would be more appropriate.

Reply:
I agree with @theoharr: from what I got from the spec, 4420 is recommended for controllers, but 8009 port is officially assigned to service discovery:

> TCP port 4420 has been assigned for use by NVMe over Fabrics and TCP port 8009 has been assigned
> by IANA (refer to https://www.iana.org/assignments/service-names-port-numbers/service-names-portnumbers.xhtml) for use by NVMe over Fabrics discovery. TCP port 8009 is the default TCP port for
> NVMe/TCP Discovery controllers. There is no default TCP port for NVMe/TCP I/O controllers. The
> Transport Service Identifier (TRSVCID) field in the Discovery Log Entry indicates the TCP port to use.

Reply:
Ok, reasonable.